source-code/core/app/src/createReactiveQuery.ts

```diff
@@ -0,0 +1,48 @@
+import type { Message, MessageQueryApi } from "@inlang/messages"
+import { ReactiveMap } from "@solid-primitives/map"
+import { createEffect } from "./solid.js"
+
+/**
+ * Creates a query API for messages.
+	@@ -9,41 +9,43 @@ import { createEffect } from "./solid.js"
+ */
+export function createReactiveQuery(messages: () => Array<Message>): MessageQueryApi {
+	//let index: ReactiveMap<string, Message>
+	const index = new ReactiveMap<string, Message>(messages().map((message) => [message.id, message]))
+
+	createEffect(() => {
+		// TODO: update index
```

Comment:
@inlang/team how should we update the map if new entries show up?

Reply:
- replace all entries?
- just add new entries?
- remove old entries?

Reply:
> replace all entries? 

If it's fast enough :D We can optimize later